ipld/blockstore/src/buffered.rs

```diff
@@ -34,70 +34,145 @@ where
     /// This will recursively traverse the cache and write all data connected by links to this
     /// root Cid.
     pub fn flush(&mut self, root: &Cid) -> Result<(), Box<dyn StdError>> {
-        write_recursive(self.base, &self.write.borrow(), root)?;
+        copy_rec(self.base, &self.write.borrow(), *root)?;
 
         self.write = Default::default();
+
         Ok(())
     }
 }
 
-/// Recursively traverses cache through Cid links.
-fn write_recursive<BS>(
+/// Given a CBOR encoded Buffer, get the type of the CBOR object.
+/// This was implemented because the CBOR library we use does not expose low
+/// methods like this, requiring us to deserialize the whole CBOR payload, which
+/// is unnecessary and quite inefficient for our usecase here.
+fn cbor_read_header_buf<B: std::io::BufRead>(
```

Comment:
could use `use std::io::BufRead` as we have two instance of it with absolute paths in this module.